include/ttmlir/Dialect/TT/IR/TTOpsEnums.td

```diff
@@ -72,6 +72,26 @@ def TT_MemorySpace : I32EnumAttr<"MemorySpace", "TT MemorySpace",
   let cppNamespace = "::mlir::tt";
 }
 
+def TT_UndefLayout : I32EnumAttrCase<"UndefLayout", 0, "undef_layout">; // For host tensors
+def TT_Interleaved : I32EnumAttrCase<"Interleaved", 1, "interleaved">;
+def TT_SingleBank : I32EnumAttrCase<"SingleBank", 2, "single_bank">;
+def TT_HeightSharded : I32EnumAttrCase<"HeightSharded", 3, "height_sharded">;
+def TT_WidthSharded : I32EnumAttrCase<"WidthSharded", 4, "width_sharded">;
+def TT_BlockSharded : I32EnumAttrCase<"BlockSharded", 5, "block_sharded">;
+
+def TT_TensorMemoryLayout : I32EnumAttr<"TensorMemoryLayout", "TT TensorMemoryLayout",
```

Comment:
@rpavlovicTT / @nobradovictt  for direct to metal path we will not be using this enum at all, in fact we should always assert it's `UndefLayout`.  It just seemed like more work/more complicated to specialize the `LayoutAttr` for TTNN in some way, esp since the optimizer will be written against TTIR dialect and this is one of the main optimizations.  I wonder if there's another way we could solve this issue though, it's definitely going to lead to confusion.  Perhaps at least we should name this enum `TTNNTensorMemoryLayout` despite it living in the TT dialect.

Reply:
The truly "proper" way to solve this would be to make LayoutAttr an interface, where we can instantiate either a TTNN based layout or a direct to metal based layout, it's just a decent size refactor for a single enum.  It's one of those things though where we could keep making exceptions and end up in a situation where we need the refactor anyway.

Reply:
Maybe we can decide on proper refactor for this when we introduce D Metal backend support to Optimizer. In the meantime as you suggested we can assert in DMetal pipeline that this is UndefLayout, I am even fine with renaming it to TTNNTensorMemoryLayout for now.

Reply:
I agree, we can go forward with renaming the attribute. However, I'd definitely refactor it as soon as we find dev cycles for it. We should strive for not leaving too much debt behind.

Reply:
Created issue #596 to track this